test/integration/resource.ts

```diff
@@ -0,0 +1,41 @@
+import {expect} from 'chai';
+import {Ketting, NeverCache} from '../../src';
+
+describe('Using resources', async () => {
+  const ketting = new Ketting('http://localhost:3000/hal1.json');
+
+  it('should do 2 distinct HTTP calls if request headers are different on concurrent calls', async () => {
+    ketting.cache = new NeverCache();
+
+    const resource = await ketting.follow('collection');
+
+    const [firstState, secondState] =
+      await Promise.all([
+        resource.get({headers: {'prefer': 'bar'}}),
+        resource.get({headers: {'prefer': 'baz'}})
+      ]);
+
+    expect(firstState.headers.get('preference-applied')).to.not.eql(secondState.headers.get('preference-applied'));
+  });
+
+  it('should produce only one HTTP call if request headers are the same on concurrent calls', async () => {
+    ketting.cache = new NeverCache();
+
+    const resource = await ketting.follow('collection');
+
+    const headers1 = new Headers();
+    headers1.append('prefer', 'bar');
+    headers1.append('prefer', 'baz');
+    const headers2 = new Headers();
+    headers2.append('prefer', 'bar');
+    headers2.append('prefer', 'baz');
+
+    const [firstState, secondState] =
+      await Promise.all([
+        resource.get({headers: headers1}),
```

Comment:
The test may fail without a delay on the server side.

Reply:
We can't add a delay via the headers, the test will fail due to the difference in request headers.

Reply:
Add the same delay to each request